File: src/leetcode/python/FirstUniqueCharacterInAString_387.py
class FirstUniqueCharacterInAString_387(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        import collections
        cnt = collections.Counter(s)
        for i in range(len(s)):
            if cnt[s[i]] == 1: return i
        return -1

File: src/leetcode/java/FirstUniqueCharacterInAString_387.java
package leetcode.java.easy;

public class FirstUniqueCharacterInAString_387 {
  public int firstUniqChar(String s) {
    int[] counter = new int[26];
    char[] carr = s.toCharArray();
    for (char c : carr) counter[c - 'a']++;
    for (int i = 0; i < carr.length; i++) {
      if (counter[carr[i] - 'a'] == 1) return i;
    }
    return -1;
  }
}
